src/complex.rs

```diff
@@ -287,6 +287,24 @@ impl<T: Clone + Float> Complex<T> {
     pub fn is_nan(self) -> bool {
         self.re.is_nan() || self.im.is_nan()
     }
+
+    /// Checks if the given complex number is infinite
+    #[inline]
+    pub fn is_infinite(self) -> bool {
+        !self.is_nan() && (self.re.is_infinite() || self.im.is_infinite())
```

Comment:
According to [Wolfram MathWorld about complex infinity](http://mathworld.wolfram.com/ComplexInfinity.html) this should be rather `self.arg().is_nan()`.